src/core.ts

```diff
@@ -470,112 +470,109 @@ export const noChange = {};
  */
 export {noChange as directiveValue};
 
-export const _isPrimitiveValue = (value: any) => value === null ||
-    !(typeof value === 'object' || typeof value === 'function');
+export const _isPrimitiveValue = (value: any) =>
+    (value === null ||
+     !(typeof value === 'object' || typeof value === 'function'));
 
+/**
+ * The Part interface represents a dynamic part of a template instance rendered
+ * by lit-html.
+ */
 export interface Part {
-  instance: TemplateInstance;
-  size?: number;
-}
+  _value: any;
 
-export interface SinglePart extends Part {
   setValue(value: any): void;
-}
 
-export interface MultiPart extends Part {
-  setValue(values: any[], startIndex: number): void;
+  commit(): void;
 }
 
-export class AttributePart implements MultiPart {
-  instance: TemplateInstance;
+export class AttributeCommitter {
   element: Element;
   name: string;
   strings: string[];
-  size: number;
-  _previousValues: any;
+  parts: AttributePart[];
+  dirty = true;
 
-  constructor(
-      instance: TemplateInstance, element: Element, name: string,
-      strings: string[]) {
-    this.instance = instance;
+  constructor(element: Element, name: string, strings: string[]) {
     this.element = element;
     this.name = name;
     this.strings = strings;
-    this.size = strings.length - 1;
-
-    this._previousValues = [];
+    this.parts = [];
+    for (let i = 0; i < strings.length - 1; i++) {
+      this.parts[i] = new AttributePart(this);
+    }
   }
 
-  protected _interpolate(values: any[], startIndex: number) {
+  protected _getValue(): any {
     const strings = this.strings;
     const l = strings.length - 1;
     let text = '';
 
     for (let i = 0; i < l; i++) {
       text += strings[i];
-      const v = getValue(this, values[startIndex + i]);
-      if (v && v !== noChange &&
-          (Array.isArray(v) || typeof v !== 'string' && v[Symbol.iterator])) {
-        for (const t of v) {
-          // TODO: we need to recursively call getValue into iterables...
-          text += t;
+      const part = this.parts[i];
+      if (part !== undefined) {
+        const v = part._value;
+        if (v != null &&
+            (Array.isArray(v) || typeof v !== 'string' && v[Symbol.iterator])) {
```

Comment:
I think this TODO was mostly incorrect, and fully incorrect now that this code is in AttributeCommitter. The values here are after AttributePart has already handled them, passing them to getValue. Directive handling is the biggest thing we get out of calling getValue, and arrays of directive don't make sense, since there's no individual part to pass to the directive. TemplateResults don't make sense for attribute parts - using them here would be essentially toString'ing a DOM tree into an attribute.